charts/gardener/templates/secret-internal-domain.yaml

```diff
@@ -13,12 +13,17 @@ metadata:
   annotations:
     dns.garden.sapcloud.io/provider: {{ ( required ".controller.internalDomain.provider is required" .Values.controller.internalDomain.provider ) }}
     dns.garden.sapcloud.io/domain: {{ ( required ".controller.internalDomain.domain is required" .Values.controller.internalDomain.domain ) }}
+    {{- if ne $domain.provider "alicloud-dns" }}
     dns.garden.sapcloud.io/hostedZoneID: {{ ( required ".controller.internalDomain.hostedZoneID is required" .Values.controller.internalDomain.hostedZoneID ) }}
+    {{- end }}
 type: Opaque
 data:
 {{- if eq .Values.controller.internalDomain.provider "aws-route53" }}
   accessKeyID: {{ ( required ".controller.internalDomain.credentials.accessKeyID is required" .Values.controller.internalDomain.credentials.accessKeyID ) | b64enc }}
   secretAccessKey: {{ ( required ".controller.internalDomain.credentials.secretAccessKey is required" .Values.controller.internalDomain.credentials.secretAccessKey ) | b64enc }}
+{{- else if eq .Values.controller.internalDomain.provider "alicloud-dns" }}
```

Comment:
Can't this be merged with line 21? Let's not duplicate this.

Reply:
No. Content in lines 22-23 are not the same as in lines 25-26.